grails-app/services/org/pih/warehouse/product/ProductPackageService.groovy

```diff
@@ -0,0 +1,62 @@
+package org.pih.warehouse.product
+
+import grails.gorm.transactions.Transactional
+import org.pih.warehouse.core.ProductPrice
+
+@Transactional
+class ProductPackageService {
+
+    ProductSupplier save(ProductPackageCommand command) {
+        setPackageData(command)
+        if (command.contractPricePrice) {
+            setContractPriceData(command.productSupplier, command.contractPricePrice, command.contractPriceValidUntil)
+        }
+        return command.productSupplier
+    }
+
+    private static void setPackageData(ProductPackageCommand command) {
+        ProductSupplier productSupplier = command.productSupplier
+        productSupplier.minOrderQuantity = command.minOrderQuantity
+
+        ProductPackage defaultProductPackage = productSupplier?.productPackages?.find {
+            it.uom == command.uom && it.quantity == command.productPackageQuantity
+        }
+
+        // If there is no product package with given unit of measure and of given quantity, create one
+        if (!defaultProductPackage) {
+            defaultProductPackage = new ProductPackage()
+            defaultProductPackage.name = "${command.uom?.code}/${command.productPackageQuantity}"
+            defaultProductPackage.description = "${command.uom?.name} of ${command.productPackageQuantity}"
+            defaultProductPackage.product = command.productSupplier?.product
+            defaultProductPackage.uom = command.uom
+            defaultProductPackage.quantity = command.productPackageQuantity
+            productSupplier.addToProductPackages(defaultProductPackage)
+            productSupplier.defaultProductPackage = defaultProductPackage
+        }
+        // If product package price is not provided, skip assigning pricing data
+        if (!command.productPackagePrice) {
+            return
+        }
```

Comment:
this would mean that if we set the price once, we can't make it null anymore. But this is how it is done currently in the import, hence I kept the current behavior.

Reply:
Actually, it can't become null, but it can still be `0` which it can't with my solution, because I check the truthiness. I'm gonna change it to `== null`